tests/nlu/classifiers/test_regex_message_handler.py

```diff
@@ -0,0 +1,266 @@
+        ) == set(
+            (item[ENTITY_ATTRIBUTE_TYPE], item[ENTITY_ATTRIBUTE_VALUE])
+            for item in entity_data
+        )
+    else:
+        assert unpacked_message.data[ENTITIES] is not None
+        assert len(unpacked_message.data[ENTITIES]) == 0
+
+
+@pytest.mark.parametrize(
+    "intent,entities,expected_intent,domain_entities",
+    [
+        ("wrong_intent", "{'entity': 1.0}", "other_intent", ["entity"]),
+        ("my_intent", "{'wrong_entity': 1.0}", "my_intent", ["other-entity"]),
+        ("wrong_intent", "{'wrong_entity': 1.0}", "other_intent", ["other-entity"]),
+        # special case: text "my_intent['entity1']" will be interpreted as the intent
+        ("wrong_entity", "['wrong_entity']", "wrong_entity", ["wrong_entity"]),
+    ],
+)
+def test_process_warns_if_intent_or_entities_not_in_domain(
+    regex_message_handler: RegexMessageHandler,
+    intent: Text,
+    entities: Optional[Text],
+    expected_intent: Text,
+    domain_entities: List[Text],
+):
+    # construct text according to pattern
+    text = INTENT_MESSAGE_PREFIX + intent  # do not add a confidence value
+    if entities is not None:
+        text += entities
+    message = Message(data={TEXT: text})
+
+    # construct domain from expected intent/entities
+    domain = Domain(
+        intents=[expected_intent],
+        entities=domain_entities,
+        slots=[],
+        responses={},
+        action_names=[],
+        forms={},
+    )
+
+    # expect a warning
+    with pytest.warns(UserWarning):
+        results = regex_message_handler.process([message], domain)
+    unpacked_message = results[0]
+
+    if "wrong" not in intent:
+        assert unpacked_message.data[INTENT][INTENT_NAME_KEY] == intent
+        if "wrong" in entities:
+            assert unpacked_message.data[ENTITIES] is not None
+            assert len(unpacked_message.data[ENTITIES]) == 0
+    else:
+        assert unpacked_message == message
+
+
+@pytest.mark.parametrize(
+    "text",
+    [
+        "some other text",
+        "text" + INTENT_MESSAGE_PREFIX,
+        INTENT_MESSAGE_PREFIX,
+        INTENT_MESSAGE_PREFIX + "@0.5",
+    ],
+)
+def test_process_does_not_do_anything(
+    regex_message_handler: RegexMessageHandler, text: Text
+):
+
+    message = Message(
+        data={TEXT: text, INTENT: "bla"},
+        features=[
+            Features(
+                features=np.zeros((1, 1)),
+                feature_type=FEATURE_TYPE_SENTENCE,
+                attribute=TEXT,
+                origin="nlu-pipeline",
+            )
+        ],
+    )
+
+    # construct domain from expected intent/entities
+    domain = Domain(
+        intents=["intent"],
+        entities=["entity"],
+        slots=[],
+        responses={},
+        action_names=[],
+        forms={},
+    )
+
+    parsed_messages = regex_message_handler.process([message], domain)
+
+    assert parsed_messages[0] == message
+
+
+# TODO: this isn't done anymore - what was that good for?
```

Comment:
mhm, good question. I think we should still add the `TEXT` to the returned message

Reply:
Why? - If used in combination with `TED`, we can run into the (edge case) that this weird text is interpreted as text. Meanwhile, `RulePolicies` don't care about the text because they work with the extracted intent.

Reply:
We're for example doing [this](https://github.com/RasaHQ/rasa/blob/dc116ec17b9b2df71c5a0f29da9cfff573a1c183/rasa/graph_components/adders/nlu_prediction_to_history_adder.py#L54). Even we if replace that with `original_message.text` we'd still have a `UserUttered` with the text which would get featurized by TED. We should in my opinion be carefully with omitting the text as I think too many things downstream depend on the text (e.g. Rasa X). 

Fully agree that this is not ideal and that we need to separate the user's raw input from the prediction instead of merging both into one event but I don't think we can fix this until we re-organize the events

Reply:
The `.get()` will return `None` and a `UserUttered` is able to cope with a text set to `None`. The policy featurization pipeline can deal with missing texts as well. So that doesn't look like a problem to me there.

> think too many things downstream depend on the text (e.g. Rasa X).
That on the other hand, sounds like a good point 😬 

I'll add it back in and just open an issue...

Reply:
@wochinge , (apart from the issue with keeping/removing text) to avoid confusion: The test below the `TODO` comment linked above shows that regex interpreter dealt with messages that didn't start with the "/" prefix and prepended that if it wasn't there. -- However in `nlu/run.py` and in `processor.py` the regex interpreter is only used if it does start with the "/" prefix. Which is why in the new version, the "/" is mandatory and not followed by a "?" in the pattern. Long story short: That linked test there will be removed anyway and I'll add a "text is still there" check to tests :D

Reply:
Makes sense to make it mandatory! 

> I'll add it back in and just open an issue...

Thanks 🙌🏻 And sorry for the caution 😞

Reply:
no problem at all ^^